lib/ansible/modules/group.py

```diff
@@ -153,7 +159,7 @@ def group_del(self):
             command_name = 'lgroupdel'
         else:
             command_name = 'groupdel'
-        cmd = [self.module.get_bin_path(command_name, True), self.name]
+        cmd = [self.module.get_bin_path(command_name, True), '-f' if self.force else None, self.name]
```

Comment:
It doesn't look like `-f` is a valid option for `lgroupdel`.

Reply:
I just realized this is modifying the base class. We should check if the option is applicable for any of the classes that override group_del, and document which platforms support the new option (if it's not all of them). Have you tested this against solaris?

This will also need [changelog](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-changelog-fragments) and [tests](https://github.com/ansible/ansible/tree/devel/test/integration/targets/group).

Reply:
The SunOS class could override group_del for backwards compatibility for solaris. Or you could perhaps add another class derived from Group that uses `-f` instead of doing it in the base class. I'm not sure if either of those would cause issues without digging into it more.

Using `force: true` with `local: true` causes an unhandled traceback since `lgroupdel` doesn't have a force option. That should be handled in the argpsec so there's an informative error that those can't be used together: https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#dependencies-between-module-options.